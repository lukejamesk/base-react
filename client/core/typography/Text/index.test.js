import React from 'react'
import { mount } from 'enzyme'
import { forEach } from 'ramda'
import { Text } from '.'

const textTypes = ['paragraph', 'bold', 'blockquote']

describe('<Text />', () => {
  forEach((type) => {
    it(`should render with: ${type}`, () => {
      const wrapper = mount(
        <Text type={type}>
          My content for
          {type}
        </Text>
      )
      expect(wrapper).toMatchSnapshot()
    })
  }, textTypes)
})
