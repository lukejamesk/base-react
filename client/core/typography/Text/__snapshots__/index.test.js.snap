// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render with: blockquote 1`] = `
<ForwardRef
  type="blockquote"
>
  <blockquote>
    My content for
    blockquote
  </blockquote>
</ForwardRef>
`;

exports[`<Text /> should render with: bold 1`] = `
<ForwardRef
  type="bold"
>
  <strong>
    My content for
    bold
  </strong>
</ForwardRef>
`;

exports[`<Text /> should render with: paragraph 1`] = `
<ForwardRef
  type="paragraph"
>
  <p>
    My content for
    paragraph
  </p>
</ForwardRef>
`;
